src/components/pages/collectRecordFormPages/CollectRecordFormPage/CollectRecordFormPage.js

```diff
@@ -87,6 +88,14 @@ function loadObservationsFromCollectRecordIntoTableState({
 
   if (isNewRecord) {
     handleAddEmptyInitialObservation()
+
+    if (persistedUnsavedObservationsTable?.length) {
```

Comment:
we also want to account for the fact that there may be persisted observation data for new records that were unsaved